### ISSUE TITLE
Game.dat file is faulty. It causes lag and some other problems.

### ISSUE THREAD

**Issue by striyker98** [TheSuperHackers/GeneralsGamePatch#657]
Hi all! I am erdemtekin (AKA moddb_dev)

I am a C&C Generals Zero Hour player since 2007 and a forum member since 2014. I've just discovered something new. I would like to write something about the lag problem in Zero Hour that has caused many people to stop playing.

Zero hour v1.0 doesn't have any lag issue. Even in the Twilight flame map, you can play against 7 hard ai without any lag. (As in base game Generals v1.08)

Also, path finding is way better on v1.0. For example, cliffs on maps don't matter in v1.0. The CPU doesn't gather vehicles at the edges of the cliffs. The units follow the path as they should. But path finding bug exits in v1.01, v1.02 , v1.03 and v1.04

There is also a massive memory leak bug in v1.01, v1.02 , v1.03 and v1.04.

The reason for these errors is the game.dat file.

In short, although EA fixed a few bugs in Zero Hour with the patches, also these patches greatly damaged the game in terms of game performance and path finding.

By comparing the "game.dat version 1.0" file with the "game.dat version 1.04" files, this error causing memory leak can be detected. As far as I know, the programming language of .dat files is Microsoft Visual C++ 6.0. We can also use Hex editor or something.

Someones should compare the 1.0 and 1.04 versions of the game.dat file and identify the differences. Then, version 1.04 of game.dat file should be fixed. (programming skills are required for this)

Please play against 7 hard ai in twilight flame map on Zero Hour v1.0 then you will understand what I mean (or consider the gameplay against 7 hard ai on the twilight flame map in vanilla generals v1.08)

**NOTE:**

The v1.0 version is Zero Hour installed from retail discs. (Do not install any patches after installing Zero Hour)

If you don't have any disc drive to use retail discs, you can download c&c generals deluxe edition images from archive.org and use "daemon tools" or something for mounting images.

**NOTE 2:**

I already did "setting permissions of Replays folder" method but it never offered smooth gameplay like v1.0. As I said, the main cause of the problem is the "game.dat" file.

**Comment by xezon** [comment]
Duplicate of TheSuperHackers/GeneralsGameCode#100

**Comment by striyker98** [comment]
@xezon Did you really read my post? It is not duplicate. The twilight flame map here is just an example.

**Comment by xezon** [comment]
We will debug pathfinding when so possible. Right now we can't and one bug report is enough for it.

**Comment by striyker98** [comment]
@xezon The article you shared is completely different from what I wrote. I understand you didn't read all of what I wrote. To summarize briefly for you, ZH v1.04's game.dat file not only causes path finding error but also performance issues and memory leak. So there are multiple issues. ZH v1.0's game.dat file does not have these problems.

I say ZH v1.0's game.dat file is OK.

**Comment by alanblack166** [comment]
This is not the patch's field of concern. This is an engine error, which is beyond the scope of this project.

**Comment by striyker98** [comment]
@alanblack166 Can't identify the source of this problem by comparing the game.dat v1.0 file with the game.dat v1.04 file?

**Comment by xezon** [comment]
We know there are memory leaks. To find cause, we need to run memory profiler once Thyme is done.

**Comment by alanblack166** [comment]
Even if you could identify, which is highly improbable, you cannot simply hex edit completely different or missing sections of code. A reverse-engineering project is underway which should resolve this once and for all.

**Comment by striyker98** [comment]
@xezon @alanblack166 

Understood. I don't want to lose hope. I hope these critical bugs will be fixed as soon as possible.

There are both v1.0 and v1.04 files in the archive. When you want to switch to v1.0, copy the v1.0 files to the ZH folder. Likewise, when you want to switch to v1.04, copy the v1.04 files to the ZH folder. This is a manual method for switching between versions.

You can test this yourself. Please play against 7 hard ai in twilight flame map in v1.0. (I repeat, the twilight flame map is an example used when testing. I think this is the most suitable map for use in testing because it is the most problematic map)

Here is the link:
https://drive.google.com/file/d/1GcNszh7qLgRmdAqzb6j6kKQ1Rg99e4UV/view?usp=sharing

**Comment by striyker98** [comment]
@ImTimK
@xezon 
@alanblack166 
@commy2 
@Stubbjax 
@MTKing4 
@Jundiyy 
@xamorish 

There are both v1.0 and v1.04 files in the archive. When you want to switch to v1.0, copy the v1.0 files to the ZH folder. Likewise, when you want to switch to v1.04, copy the v1.04 files to the ZH folder. This is a manual method for switching between versions. You don't need any installation.

Don't look at "Options" in the menu. The version written there is determined by the registry.

You can test this yourself. Please play against 7 hard ai in twilight flame map in v1.0. (I repeat, the twilight flame map is an example used when testing. I think this is the most suitable map for use in testing because it is the most problematic map)

Here is the link:
https://drive.google.com/file/d/1GcNszh7qLgRmdAqzb6j6kKQ1Rg99e4UV/view?usp=sharing

**Comment by striyker98** [comment]
@ImTimK
@xezon 
@alanblack166 
@commy2 
@Stubbjax 
@MTKing4 
@Jundiyy 
@xamorish 
@Generalcamo

There are both v1.0 and v1.04 files in the archive. When you want to switch to v1.0, copy the v1.0 files to the ZH folder. Likewise, when you want to switch to v1.04, copy the v1.04 files to the ZH folder. This is a manual method for switching between versions. You don't need any installation.

Don't look at "Options" in the menu. The version written there is determined by the registry.

You can test this yourself. Please play against 7 hard ai in twilight flame map in v1.0. (I repeat, the twilight flame map is an example used when testing. I think this is the most suitable map for use in testing because it is the most problematic map)

Here is the link:
https://drive.google.com/file/d/1GcNszh7qLgRmdAqzb6j6kKQ1Rg99e4UV/view?usp=sharing

**Comment by alanblack166** [comment]
Stop spamming everyone. This is beyond the project's scope; projects that concern the engine will resolve this.

**Comment by striyker98** [comment]
@ImTimK
@alanblack166
@xezon 
@commy2 
@Stubbjax 
@MTKing4 
@Jundiyy 
@xamorish 
@Generalcamo
@OmniBlade
@jonwil
@feliwir

This is not spam. I am editing my last comment. Also, this is a notification to the people.

I know thyme is related to this issue. (because of engine problem)

Anyway, there are both v1.0 and v1.04 files in the archive. When you want to switch to v1.0, copy the v1.0 files to the ZH folder. Likewise, when you want to switch to v1.04, copy the v1.04 files to the ZH folder. (This is a manual method for switching between versions. You don't need any installation)

Don't look at "Options" in the menu. The version written there is determined by the registry.

You can test this yourself. Please play against 7 hard ai in twilight flame map in v1.0. (I repeat, the twilight flame map is an example used when testing. I think this is the most suitable map for use in testing because it is the most problematic map)

Here is the link:
https://drive.google.com/file/d/1GcNszh7qLgRmdAqzb6j6kKQ1Rg99e4UV/view?

**Comment by ZekeDlyoung** [comment]
> This is not spam. I am editing my last comment. Also, this is a notification to the people.

Whatever you call it, stop it, it's very annoying
